Fixing bytea json conversion error (update to hibernate 6, spring 3)

kinow opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Closes #664

WIP: will fill in rest later, and move commits to separate pull requests. Just removing warnings while I investigate the issue in #664.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

kinow commented:
I think I got all the changes out, and merged. Those were mainly deprecation warnings that I found while trying to run the code, and a missing header. Now wait for GH Actions, then I will perform one final test. Fingers-crossed it will pass, and then it will be ready for someone to review. :crossed_fingers:

kinow commented:
It worked! To review, one can confirm the code looks alright, GH Actions build is passing, and maybe try `docker compose up` after altering the docker compose file to use the `build: .` so that it builds a new image for the SpringBoot service, instead of using the latest image deployed.

@martinchapman I asked @mr-c to review it, but it would be great if you could try it too, if you have time (or maybe after this gets merged too), to confirm the issue is gone and the viewer is now working for you.

Cheers

mr-c commented:
Thank you very much @kinow ! Do you know why this wasn't caught by the CI? Do we need a new test?

kinow commented:
> Thank you very much @kinow ! Do you know why this wasn't caught by the CI? Do we need a new test?

Not a problem, @mr-c :slightly_smiling_face: 

Good point! I think we didn't have a test for retrieving a workflow (or we did, but not necessarily for the part of the code where this happened), only for the queuedworkflow.

I'll write a test once I have time to work on this again. That should give @chapmanb time to take a look too.

Thanks!

kinow commented on src/main/java/org/commonwl/view/workflow/WorkflowRepositoryImpl.java:
```diff
@@ -0,0 +1,51 @@
+
+    query.setParameter("retrievedFrom", retrievedFrom, new JsonType(GitDetails.class));
+    return (Workflow) query.uniqueResult();
+  }
```
Note to self: write a test for this!

kinow commented:
Fixed last warning (it was complaining about our old version of postgres -- 9, while the latest is 17), and added a unit test.

Note, there was one test consistently failing for me, due to a timeout to retrieve spdx licenses. I wonder whether it's a IP block, or maybe only GH Action/AWS/etc are allowed. But if that test fails, it can be ignored as it's not caused by any of the recent changes (you can try `curl -XGET https://....` to fetch the same URL and you should get a connection timed out error as well).

mr-c commented on src/main/java/org/commonwl/view/workflow/WorkflowRepositoryCustom.java:
```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
```
cwlviewer is not part of Apache, so perhaps we shouldn't say this?

kinow replied:
+1. Huh, I think I copied the header from another file in the project. I hope I didn't accidentally copy-paste from a file outside the project (or a file that was copied from an AL project).

mr-c commented:
Thank you @kinow !
